overviews/collections/iterators.md

```diff
@@ -174,3 +174,24 @@ Every iterator can be converted to a buffered iterator by calling its `buffered`
     res11: Int = 2
 
 Note that calling `head` on the buffered iterator `bit` does not advance it. Therefore, the subsequent call `bit.next()` returns the same value as `bit.head`.
+
+**Warning**: From the `buffered` [API documentation](http://www.scala-lang.org/api/current/index.html#scala.collection.Iterator):
+
+> Reuse: After calling this method, one should discard the iterator it was called on,
+> and use only the iterator that was returned. Using the old iterator is undefined,
+> subject to change, and may result in changes to the new iterator as well.
+
+You might be tempted to access the first element in the iterator `it` by calling `it.buffered.head` and then continue manipulating `it`.  Instead you should assign `it.buffered` to a variable, and use the result.
+
+For example, if you wish to pass an iterator and its first element to a function, call the function as follows:
+
+    // Correct
+    val it: Iterator[Int] = ...
+    val bit = it.buffered
+    myFunction(bit.head, bit)
+
+    // Incorrect
+    val it: Iterator[Int] = ...
+    myFunction(it.buffered.head, it)
```

Comment:
in #551 I just reiterate, "don't re-use the original iterator", which is the theme of most of the page.

I also place an ad for `Iterator.single(head) ++ rest`. It makes me wonder if `BufferedIterator.toIterator` should do that; except that it doesn't have much overhead, which is another point I added.